pkg/scheduler/scheduler.go

```diff
@@ -315,9 +318,24 @@ func (s *Scheduler) processAllocationUpdateEvent(ev *schedulerevent.SchedulerAll
 	if len(ev.AcceptedAllocations) > 0 {
 		alloc := ev.AcceptedAllocations[0]
 		// Update pending resource
-		if err := s.confirmAllocationProposal(alloc); err != nil {
+		if err := s.confirmAllocationProposal(alloc.AllocationProposal); err != nil {
 			log.Logger().Error("failed to confirm allocation proposal",
 				zap.Error(err))
+			// if the scheduler could not confirm the proposal,
+			// we need to notify the cache to revert the allocations
+			for _, allocInfo := range alloc.Allocations {
+				s.eventHandlers.CacheEventHandler.HandleEvent(&cacheevent.ReleaseAllocationsEvent{
```

Comment:
Since Once Cache received ReleaseAllocationsEvent, Cache will notify RMProxy: 

```
			m.notifyRMAllocationReleased(rmID, releasedAllocations, toReleaseAllocation.ReleaseType, toReleaseAllocation.Message)
```

For this case (scheduler cannot confirm allocation), I think we should skip notifying RM, because this is not an allocation RM aware of.

Reply:
Currently, shim has no action to take for released allocations.
But this makes sense, it's better to add another `ReleaseType` in the scheduler-interface for this case, maybe`AllocationReleaseResponse_REJECTED_BY_SCHEDULER `? We can track this enhancement in a separate JIRA.

Reply:
I don't suggest to change scheduler interface for the purpose. scheduler-interface is supposed to be used for communication between core and shim, not within the shim. 

Or we can add a field to indicate the allocation is a "transient allocation" so cache can find there's no need to send the information back to RMProxy.

I think it is fine to have a separate JIRA because notifying RM about the extra released resource should be fine. But let's fix it by 0.9 release (mark priority to be blocker).